src/main/java/com/terraformersmc/terrestria/mixin/MixinHoeItem.java

```diff
@@ -0,0 +1,50 @@
+package com.terraformersmc.terrestria.mixin;
+
+import com.google.common.collect.ImmutableMap;
+import com.terraformersmc.terrestria.init.TerrestriaBlocks;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.HoeItem;
+import net.minecraft.item.ItemUsageContext;
+import net.minecraft.sound.SoundCategory;
+import net.minecraft.sound.SoundEvents;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+import net.minecraft.world.World;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.ModifyConstant;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Mixin(HoeItem.class)
+public class MixinHoeItem {
```

Comment:
This mixin would preferably be replaced by methods on the block entirely, with a common method